

Rust używa typów `Result` i `Option`, aby pomóc w obsłudze błędów w kodzie. Mechanizmy te są bezpieczniejsze niż wyjątki.

**Przykład użycia** `Result`:
```rust
fn divide(dividend: f64, divisor: f64) -> Result<f64, String> {
    if divisor == 0.0 {
        Err("Division by zero".to_string())
    } else {
        Ok(dividend / divisor)
    }
}

fn main() {
    match divide(10.0, 2.0) {
        Ok(result) => println!("Result: {}", result),
        Err(e) => println!("Error: {}", e),
    }
}

```






